Add inforamtion about new deploy action (#1641)
astronomer/docs: astro/ci-cd.md

- Multiple branch:  Deploys multiple branches to separate Deployments on Astro.
- Custom image:  Deploys an Astro project with a customized Runtime image and additional build arguments.

## Deploy with GitHub Actions

> The following templates use Astronomer's [Deploy Action](https://github.com/astronomer/deploy-action) to deploy code to Astronomer. The Deploy Action is the easiest way to create CI/CD pipelines with Astro. Take a look at the [Deploy Action readme](https://github.com/astronomer/deploy-action#readme) to learn more about its many input options.

Reviewer: ```suggestion
The following templates use Astronomer's [Deploy Action](https://github.com/astronomer/deploy-action) to deploy code to Astronomer. The Deploy Action is the easiest way to create CI/CD pipelines for Astro. See the [Deploy Action README](https://github.com/astronomer/deploy-action#readme) to learn more about using and customizing this action.
```

### GitHub Actions (DAG-based deploy)

> The following templates are examples of how to implement DAG-only deploys in GitHub Actions. They use the [Deploy Action's](https://github.com/astronomer/deploy-action) `dag-deploy-enabled` option to implement a dag-based deploy workflow.

Reviewer: ```suggestion
The following templates are examples of how to implement DAG-only deploys in GitHub Actions. They use the [Deploy Actions](https://github.com/astronomer/deploy-action) `dag-deploy-enabled` option to implement a DAG-based deploy workflow.
```
